### ISSUE TITLE
tomcat context thread leak

### ISSUE THREAD

**Issue by jjimjam** [atomikos/transactions-essentials#127]
I get a thread leak when stopping a tomcat context.   Perhaps not a serious problem, but I'd like to check anyway.

This is from the tomcat log:

06-Oct-2020 11:56:23.024 WARNING [http-nio-3030-exec-6] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [mywebapp] appears to have started a thread named [Atomikos:2] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.base@14/jdk.internal.misc.Unsafe.park(Native Method)
 java.base@14/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
 java.base@14/java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:462)
 java.base@14/java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:361)
 java.base@14/java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:937)
 java.base@14/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1055)
 java.base@14/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1116)
 java.base@14/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
 java.base@14/java.lang.Thread.run(Thread.java:832)

 
The reason for this is that com.atomikos.thread.TaskManager.AtomikosThreadFactory creates threads with the current thread's classloader. 
	thread.setContextClassLoader(Thread.currentThread().getContextClassLoader());

For webapp datasources this is the webapp's classloader.   But the TaskManger SINGLETON is on the Server's class loader because I started the UserTransactionManager with a Server Lifecyclelistener.  When the tomcat context (and all its datasources) are closed, the TaskManager threads remain in the pool.  These still use the webapp's classloader.   

I have been changing TaskManger to use the main classloader.
	thread.setContextClassLoader(getClass().getClassLoader());

It appears to work, but I'm not 100% certain of the impact.

**Comment by GuyPardon** [comment]
Good catch! At first sight I am not sure of the impact either but we'll have a look.
